ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/diffbased/common/worldview/accumulator/preload/CodeConsumingMap.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright contributors to Hyperledger Besu.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.trie.diffbased.common.worldview.accumulator.preload;
+
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.ethereum.trie.diffbased.common.DiffBasedValue;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentMap;
+import javax.annotation.Nonnull;
+
+import com.google.common.collect.ForwardingMap;
+import org.apache.tuweni.bytes.Bytes;
+
+/**
+ * A map that stores Ethereum addresses and their associated code values, with a consumer to process
+ * updates.
+ *
+ * <p>Each time a new value is added or updated, the consumer processes the code value.
+ *
+ * <p>This class uses a thread-safe {@link ConcurrentMap} to store data, allowing concurrent
+ * modifications.
+ */
+public class CodeConsumingMap extends ForwardingMap<Address, DiffBasedValue<Bytes>> {
```

Comment:
It looks like `AccountConsumingMap` is almost a complete copy of this class and you can achieve the same reusing that class. Why not renaming `AccountConsumingMap` -> `ConsumingMap` and declare it like:
```
codeToUpdate = new ConsumingMap<Bytes>(new ConcurrentHashMap<>(), codePreloader);
```

Reply:
it's not completly the same, there is a difference in the consumer.process part (put method)

Reply:
ah true! missed that, in `CodeConsumingMap` only the updated value is processed by the consumer. But is there any reason for the Consumer not to do that itself and only use the updated value, rather than doing it inside the `CodeConsumingMap` ?

Reply:
not really , more because I didn't wanted to push the logic here

```java
public DiffBasedWorldStateUpdateAccumulator(
      final DiffBasedWorldView world,
      final Consumer<DiffBasedValue<ACCOUNT>> accountPreloader,
      final Consumer<StorageSlotKey> storagePreloader,
      final Consumer<Bytes> codePreloader,
      final EvmConfiguration evmConfiguration) {
    super(world, evmConfiguration);
    this.accountsToUpdate = new AccountConsumingMap<>(new ConcurrentHashMap<>(), accountPreloader);
    this.codeToUpdate = new CodeConsumingMap(new ConcurrentHashMap<>(), codePreloader);
    this.accountPreloader = accountPreloader;
    this.storagePreloader = storagePreloader;
    this.codePreloader = codePreloader;
    this.isAccumulatorStateChanged = false;
    this.evmConfiguration = evmConfiguration;
  ```
  but I don't have a strong opinion about that